demo/admin/src/products/categories/AssignProductsGrid.tsx

```diff
@@ -0,0 +1,78 @@
+import { gql, useApolloClient } from "@apollo/client";
+import { Savable } from "@comet/admin";
+import {
+    GQLGetProductIdsForProductCategoryQuery,
+    GQLGetProductIdsForProductCategoryQueryVariables,
+    GQLSetProductCategoryMutation,
+    GQLSetProductCategoryMutationVariables,
+} from "@src/products/categories/AssignProductsGrid.generated";
+import { ProductsSelectGrid } from "@src/products/categories/ProductsSelectGrid";
+import isEqual from "lodash.isequal";
+import React, { useEffect, useState } from "react";
+
+const setProductCategoryMutation = gql`
+    mutation SetProductCategory($id: ID!, $input: [ID!]!) {
+        updateProductCategory(id: $id, input: { products: $input }) {
+            id
+        }
+    }
+`;
+
+const getProductIdsForProductCategory = gql`
+    query GetProductIdsForProductCategory($id: ID!) {
+        products(filter: { category: { equal: $id } }) {
+            nodes {
+                id
+            }
+        }
+    }
+`;
+
+interface FormProps {
+    productCategoryId: string;
+}
+
+export function AssignProductsGrid({ productCategoryId }: FormProps): React.ReactElement {
+    const client = useApolloClient();
+
+    const [selection, setSelection] = useState<{ initialValues?: string[]; values: string[] }>({ values: [] });
```

Comment:
please split that into two states. And for initialValues I would use useQuery

Reply:
I tried this in my first attempt, but I had no idea how to use data from useQuery as initial-value for values-state in a clean way. what do you think about this:
```
    const { data, error, loading } = useQuery<GQLGetProductIdsForProductCategoryQuery, GQLGetProductIdsForProductCategoryQueryVariables>(
        getProductIdsForProductCategory,
        {
            variables: { id: productCategoryId },
        },
    );
    const initialValues = data?.products.nodes ? data.products.nodes.map((product) => product.id).sort() : [];
    const [values, setValues] = useState<string[]>(initialValues);
    useEffect(() => {
        setValues(initialValues);
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [data?.products.nodes]);

    if (error) return <FormattedMessage id="common.error" defaultMessage="An error has occured. Please try again at later" />;
    if (loading) return <CircularProgress />;
```

okay, now that I see it I think it's ok. Do you know any trick to avoid the disable-react-hooks/exhaustive-deps? using initialValues directly does obviously always trigger a state-change...